tests/behavior_test.cpp

```diff
@@ -0,0 +1,54 @@
+#include "catch/catch.hpp"
+
+#include "behavior.h"
+
+TEST_CASE( "behavior_tree", "[behavior]" )
+{
+    bool is_cold = true;
+    bool is_thirsty = true;
+    bool is_hungry = true;
+
+    behavior::node cold;
+    cold.set_goal( "get_warm" );
+    cold.set_predicate( [&]( const Creature * ) {
+        return is_cold ? behavior::running : behavior::success;
+    } );
+    behavior::node thirst;
+    thirst.set_goal( "get_water" );
+    thirst.set_predicate( [&]( const Creature * ) {
+        return is_thirsty ? behavior::running : behavior::success;
+    } );
+    behavior::node hunger;
+    hunger.set_goal( "get_food" );
+    hunger.set_predicate( [&]( const Creature * ) {
+        return is_hungry ? behavior::running : behavior::success;
+    } );
+
+    behavior::node basic_needs;
+    basic_needs.set_strategy( &behavior::default_sequential );
```

Comment:
There's no test of `behavior::default_fallback` so far.  It should be added for full coverage.

Reply:
Belated, it also would be good (but not required) to add another level to the tree.  I don't see any reason why this code wouldn't handle a n+1 level tree, but if we don't have to guess, we don't have to be worry about being wrong.

Reply:
Will do both, I partially just ran out of patience typing entries in since I had mislaid my behavior sketch.